eth/peer.go

```diff
@@ -285,13 +296,19 @@ func (p *peer) readStatus(network uint64, status *statusData, genesis common.Has
 
 	p.Log().Proto("<<"+ethCodeToString[msg.Code], "receivedAt", unixTime, "obj", status, "size", int(msg.Size), "peer", p.ID())
 
+	statusWrapper.ReceivedAt = &msg.ReceivedAt
+	statusWrapper.Status = &status
+
 	if status.GenesisBlock != genesis {
```

Comment:
@zzma I decided to keep this behavior (as well as checking the error code again in `eth/handler.go:216`) for now. If these 3 mismatch checks are removed, our node will assume the peer is compatible (e.g. on the same blockchain) and resume other behaviors, such as DAO fork check through BlockHeader exchange. I assume the peer will do the same checks and disconnect us anyways, but I am not sure if that assumption is enough to remove the checks.